src/haproxy.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""The haproxy service module."""
+
+from charms.operator_libs_linux.v0 import apt
+from charms.operator_libs_linux.v1 import systemd
+
+APT_PACKAGE_VERSION = "2.8.5-1ubuntu3"
+APT_PACKAGE_NAME = "haproxy"
+
+
+class HAProxyService:
+    """HAProxy service class.
+
+    Attrs:
+       is_active: Indicate if the haproxy service is active and running.
+    """
+
+    def install(self) -> None:
+        """Install the haproxy apt package."""
+        apt.update()
+        apt.add_package(package_names=APT_PACKAGE_NAME, version=APT_PACKAGE_VERSION)
+
+    @property
```

Comment:
I'm not sure if this check should be a property.

Reply:
Indeed, I updated this to a method, I'll also add a retry library in a follow-up PR